### PR TITLE
added dispute

### PR DESCRIPTION
**Pull request by ankitdas13**
Doc reference

[Fetch All Disputes](https://razorpay.com/docs/api/disputes/fetch-all)
[Fetch a Dispute](https://razorpay.com/docs/api/disputes/fetch)
[Accept a Dispute](https://razorpay.com/docs/api/disputes/accept)
[Contest a Dispute](https://razorpay.com/docs/api/disputes/contest)

### CODE REVIEW

File: documents/dispute.md [hunk 1]
```diff
@@ -0,0 +1,286 @@
+## Document
+
+### Fetch All Disputes
+
```
Comment (bhavyay) [inline documents/dispute.md:4]: Can you provide API docs in the PR description ?

File: documents/dispute.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+```rb
+Razorpay::Dispute.all()
```
Comment (bhavyay) [inline documents/dispute.md:6]: Do we allow any params for pagination ?

File: documents/dispute.md [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+
+```rb
+disputeId = "disp_0000000000000"
+
+Razorpay::Dispute.fetch(disputeId)
+```
+
+**Parameters:**
+
+| Name  | Type      | Description                                      |
```
Comment (bhavyay) [inline documents/dispute.md:59]: please format the table

File: documents/dispute.md [hunk 4]
```diff
@@ -0,0 +1,286 @@
...
+    }
+  ],
+  "action": "draft"
+})
+```
+
+**Parameters:**
+
+| Name  | Type      | Description                                      |
+|-------|-----------|--------------------------------------------------|
```
Comment (bhavyay) [inline documents/dispute.md:174]: please format the table

File: documents/dispute.md [hunk 5]
```diff
@@ -0,0 +1,286 @@
...
+    "refund_cancellation_policy": null,
+    "term_and_conditions": null,
+    "others": null,
+    "submitted_at": null
+  }
+}
+```
+-------------------------------------------------------------------------------------------------------
+
+### Fetch a Dispute
```
Comment (bhavyay) [inline documents/dispute.md:97]: can you update the title for this section ?

File: documents/dispute.md [hunk 6]
```diff
@@ -0,0 +1,286 @@
...
+    "doc_EFtmUsbwpXwBG9",
+    "doc_EFtmUsbwpXwBG8"
+  ],
+  "action": "submit"
+})
+```
+
+**Response:**
+```json
+// Draft
```
Comment (bhavyay) [inline documents/dispute.md:195]: Not clear on why this is added. Can you add more details if this is a specific case of response

File: test/razorpay/test_dispute.rb [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+    end
+
+    def test_fetch_all_dispute
+      stub_get(/disputes$/, 'dispute_collection')
+      dispute = Razorpay::Dispute.all
+      assert_instance_of Razorpay::Collection, dispute, 'Dispute should be an array'
+      refute_empty dispute.items, 'Dispute should be more than one'
+    end
+
+    def test_dispute_accept
```
Comment (bhavyay) [inline test/razorpay/test_dispute.rb:33]: please add test cases for failure cases of these APIs.

File: documents/dispute.md (hunk 1 above)
Comment (ankitdas13) [reply documents/dispute.md:4]: Pr description has been updated

File: documents/dispute.md (hunk 2 above)
Comment (ankitdas13) [reply documents/dispute.md:6]: No params according to this doc https://razorpay.com/docs/api/disputes/fetch-all

File: documents/dispute.md (hunk 3 above)
Comment (ankitdas13) [reply documents/dispute.md]: Done

File: documents/dispute.md (hunk 4 above)
Comment (ankitdas13) [reply documents/dispute.md]: Done

File: documents/dispute.md (hunk 6 above)
Comment (ankitdas13) [reply documents/dispute.md]: Separated both code snippets `submit` or `draft`

File: test/razorpay/test_dispute.rb (hunk 7 above)
Comment (ankitdas13) [reply test/razorpay/test_dispute.rb:43]: Added failure test cases

File: documents/dispute.md [hunk 8]
```diff
@@ -0,0 +1,299 @@
...
+}
+```
+
+-------------------------------------------------------------------------------------------------------
+### Contest a Dispute (Submit)
+
+```rb
+# Use this API sample code for submit
+
+Razorpay::Dispute.fetch(disputeId).contest({
```
Comment (bhavyay) [inline documents/dispute.md:232]: can you update this ?

Comment (ankitdas13) [reply documents/dispute.md]: updated

https://github.com/razorpay/razorpay-ruby/blob/2e0b54f61b54b10dff29bb1d178a7fcf95b5effe/documents/dispute.md?plain=1#L234

**Review by bhavyay — APPROVED** [review]

**Pull request merged**